### DISCUSSION TITLE
TSML-UI edit Missing

### DISCUSSION THREAD

**Discussion by Grateful-Viv** [code4recovery/12-step-meeting-list#843] [Q&A] [answered]
For some reason my live site is not able to show TSML-UI edit. It's missing from the admin bar at the top. It shows up fine on the staging sites.

I have installed the latest TSML plugin version (3.14.4) and have it set to run TSML-UI, but even opening the Java Script console within a meeting doesn't show the edit. 

Only two lines are displayed.

JQMIGRATE: Migrate is installed, version 3.3.2
app.js?ver=6.0:2 TSML UI meeting finder: https://github.com/code4recovery/tsml-ui

Any idea what I can do? This is the actual site I would like to edit meetings on.  I have already checked the active plugins on the live site against the staging sites and they are the same.

**Comment by joshreisner** [comment, marked as answer]
i see it in the console on your production site, is that what you were talking about?

<img width="1466" alt="Screen Shot 2022-06-26 at 5 29 12 PM" src="https://user-images.githubusercontent.com/1551689/175840292-996003d5-f483-4890-94a5-19e4a0f1e77e.png">

**Reply by Grateful-Viv** [reply]
It's the weirdest thing. It wasn't there at all. I was about to rollback to a different version of the plugin, but first thought to do a deeper dive on the plugins. I ended up deactivating plugins one by one, then testing to see if it would work and saw no change. Then I went in through the admin dashboard to Meetings, selected one to edit and updated it. When I went to preview meeting, I saw the Edit Meeting on the Toolbar. Also the JavaScript console (Shift CTRL J) then showed the TSML UI Edit option for the meeting. It's like the Edit option was woken up.

But, it's also showing previous meetings I've opened. Is this normal? This only occurs through the console, not when I select Edit from the admin toolbar.

![Edit Screen](https://user-images.githubusercontent.com/62247421/175841114-08e2295d-0ca1-477e-b0fe-2c53c1de3c9b.png)

**Reply by joshreisner** [reply]
ah good. yes its normal to see all the meetings. the log isn't cleared unless you actually refresh the page or navigate to a totally different page

**Reply by Grateful-Viv** [reply]
This happened again after updating plugin to v3.14.5 

I am using TSML-UI as the front end.

When I select a meeting from the front end, Edit is missing. Open the JavaScript console and edit meeting is missing.

Manual steps to correct:

1. In WP Admin, go to Meetings and select a meeting. 
2. Do any edit and UPDATE.
3. Select [View meeting] from Meeting updated. 
4. Bingo! Edit Meeting option is now present.

**Reply by Grateful-Viv** [reply]
Just updated TSML to  3.14.7, and had the same thing happen. There was no edit option in the admin tool bar at the top, and the JavaScript console did not display it either. I did the manual steps as outline above, and got the edit button back on the toolbar. Does this happen to anyone else?

**Reply by joshreisner** [reply]
mm yes i see the issue on our demo site as well. seems like updating the plugin somehow wipes out the `edit_url` and then editing brings it back

created an issue: https://github.com/code4recovery/12-step-meeting-list/issues/972
